BUILD: Wheel building is failing on main

Starting with the latest [scheduled wheel build](https://github.com/numpy/numpy/actions/runs/2465556423), we are getting errors from cibuildwheel tests in loadtxt.py:
```
022-06-09T06:34:28.5128228Z FAILED lib/tests/test_loadtxt.py::test_integer_signs[B] - Failed: DID NOT RAI...
2022-06-09T06:34:28.5128658Z FAILED lib/tests/test_loadtxt.py::test_integer_signs[H] - Failed: DID NOT RAI...
2022-06-09T06:34:28.5129069Z FAILED lib/tests/test_loadtxt.py::test_integer_signs[I] - Failed: DID NOT RAI...
2022-06-09T06:34:28.5129476Z FAILED lib/tests/test_loadtxt.py::test_integer_signs[L] - Failed: DID NOT RAI...
2022-06-09T06:34:28.5129903Z FAILED lib/tests/test_loadtxt.py::test_integer_signs[Q] - Failed: DID NOT RAI...
2022-06-09T06:34:28.5130313Z FAILED lib/tests/test_loadtxt.py::test_integer_signs[P] - Failed: DID NOT RAI...
2022-06-09T06:34:28.5130729Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[b] - ...
2022-06-09T06:34:28.5131136Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[B] - ...
2022-06-09T06:34:28.5131717Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[h] - ...
2022-06-09T06:34:28.5132134Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[H] - ...
2022-06-09T06:34:28.5132539Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[i] - ...
2022-06-09T06:34:28.5132949Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[I] - ...
2022-06-09T06:34:28.5133342Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[l] - ...
2022-06-09T06:34:28.5133850Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[L] - ...
2022-06-09T06:34:28.5134253Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[q] - ...
2022-06-09T06:34:28.5134657Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[Q] - ...
2022-06-09T06:34:28.5135059Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[p] - ...
2022-06-09T06:34:28.5135465Z FAILED lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails[P] - ...
```

Maybe something to do with #21688 ?

--- a/numpy/core/tests/test_deprecations.py
+++ b/numpy/core/tests/test_deprecations.py
@@ -1216,6 +1216,12 @@ class TestAxisNotMAXDIMS(_DeprecationTestCase):
 
 
 class TestLoadtxtParseIntsViaFloat(_DeprecationTestCase):
+    # Deprecated 2022-07-03, NumPy 1.23
+    # This test can be removed without replacement after the deprecation.
+    # The tests:
+    #   * numpy/lib/tests/test_loadtxt.py::test_integer_signs
+    #   * lib/tests/test_loadtxt.py::test_implicit_cast_float_to_int_fails
+    # Have a warning filter that needs to be removed.
     message = r"loadtxt\(\): Parsing an integer via a float is deprecated.*"
 
     @pytest.mark.parametrize("dtype", np.typecodes["AllInteger"])

--- a/numpy/lib/tests/test_loadtxt.py
+++ b/numpy/lib/tests/test_loadtxt.py
@@ -390,6 +390,7 @@ def test_bool():
 @pytest.mark.skipif(IS_PYPY and sys.implementation.version <= (7, 3, 8),
                     reason="PyPy bug in error formatting")
 @pytest.mark.parametrize("dtype", np.typecodes["AllInteger"])
+@pytest.mark.filterwarnings("error:.*integer via a float.*:DeprecationWarning")
 def test_integer_signs(dtype):
     dtype = np.dtype(dtype)
     assert np.loadtxt(["+2"], dtype=dtype) == 2
@@ -407,6 +408,7 @@ def test_integer_signs(dtype):
 @pytest.mark.skipif(IS_PYPY and sys.implementation.version <= (7, 3, 8),
                     reason="PyPy bug in error formatting")
 @pytest.mark.parametrize("dtype", np.typecodes["AllInteger"])
+@pytest.mark.filterwarnings("error:.*integer via a float.*:DeprecationWarning")
 def test_implicit_cast_float_to_int_fails(dtype):
     txt = StringIO("1.0, 2.1, 3.7\n4, 5, 6")
     with pytest.raises(ValueError):

